openedx/core/djangoapps/discussions/docs/decisions/0007-topic-context-for-deletion.rst

```diff
@@ -0,0 +1,80 @@
+Store more context for topics to allow more flexible handling of deletion
+=========================================================================
+
+
+Status
+------
+
+Proposal
+
+
+Context
+-------
+
+The ``DiscussionTopicLink`` model currently stores all relevant information for
+topics in a course. It can handle both course-wide topics and in-context topics.
+Currently when used for in-context discussions, each topic is linked to a usage
+key that defines the context in which it is used. 
+
+However, the usage key itself is no longer sufficient once the content it points
+to has been deleted. For instance, if a Unit is deleted, the usage key no longer
+provides much meaningful information other than to indicate that the topic was 
+at one point existed in a course context. Likewise if a section or subsection is
+deleted. 
+
+Once the unit the usage key for an in-context topic points to no longer exists 
+in the course, there is no way to indicate to moderators or course authors where
+it belonged historically. 
+
+Requirements
+------------
+
+When course content is deleted, we need to be able to save the original context
```

Comment:
Can you elaborate on what is `original context` is referred to here?

Reply:
Will do.